### ISSUE TITLE
[Bug]: E5108: Error executing lua after upgrade to nvim 0.10.0

### ISSUE THREAD

**Issue by artemlive** [akinsho/bufferline.nvim#903]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When neovim starts, I got the error:
```
E5108: Error executing lua ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:119: Segments must be a list
stack traceback:
        [C]: in function 'assert'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:119: in function 'get_component_size'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:480: in function <...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:461>
        vim/shared.lua: in function 'get_components'
        ...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:56: in function <...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:54>
```

### What did you expect to happen?

I expected it to work. 

### Config

```
return {
  'akinsho/bufferline.nvim',
  event = 'VeryLazy',
  keys = {
    { '<leader>bp', '<Cmd>BufferLineTogglePin<CR>', desc = 'Toggle Pin' },
    { '<leader>bP', '<Cmd>BufferLineGroupClose ungrouped<CR>', desc = 'Delete Non-Pinned Buffers' },
    { '<leader>bo', '<Cmd>BufferLineCloseOthers<CR>', desc = 'Delete Other Buffers' },
    { '<leader>br', '<Cmd>BufferLineCloseRight<CR>', desc = 'Delete Buffers to the Right' },
    { '<leader>bl', '<Cmd>BufferLineCloseLeft<CR>', desc = 'Delete Buffers to the Left' },
    --    { "<S-h>", "<cmd>BufferLineCyclePrev<cr>", desc = "Prev Buffer" },
    --    { "<S-l>", "<cmd>BufferLineCycleNext<cr>", desc = "Next Buffer" },
    { '[b', '<cmd>BufferLineCyclePrev<cr>', desc = 'Prev Buffer' },
    { ']b', '<cmd>BufferLineCycleNext<cr>', desc = 'Next Buffer' },
  },
  opts = {
    options = {
      -- stylua: ignore
      close_command = function(n) require("mini.bufremove").delete(n, false) end,
      -- stylua: ignore
      right_mouse_command = function(n) require("mini.bufremove").delete(n, false) end,
      diagnostics = 'nvim_lsp',
      always_show_bufferline = true,
      offsets = {
        {
          filetype = 'neo-tree',
          text = 'Neo-tree',
          highlight = 'Directory',
          text_align = 'left',
        },
      },
    },
  },
  config = function(_, opts)
    local wk = require 'which-key'

    -- Define your custom buffer close function
    local close_buffer = function()
      require('mini.bufremove').delete(vim.fn.bufnr(), false)
    end

    wk.register({
      b = {
        name = '[B]uffers', -- Group name for buffer-related commands
        c = { close_buffer, 'Close the buffer' }, -- Your custom command with description
      },
    }, { prefix = '<leader>' })

    require('bufferline').setup(opts)
    -- Fix bufferline when restoring a session
    vim.api.nvim_create_autocmd('BufAdd', {
      callback = function()
        vim.schedule(function()
          pcall(nvim_bufferline)
        end)
      end,
    })
  end,
}

```

### Additional Information

After the upgrading to nvim 0.10.0 bufferline doesn't work. 

### commit
f4b4b98
_No response_

**Comment by ndom91** [comment]
Just ran into this too using:

- `bufferline@4.5.3` (`f4b4b98`)
- `nvim@v0.10.0-dev-9ca81b0`

**Comment by ecosse3** [comment]
Same issue here on 0.11 dev.

**Comment by ndom91** [comment]
[This fork](https://github.com/Theyashsawarkar/bufferline.nvim) mentioned in this issue (https://github.com/akinsho/bufferline.nvim/issues/895) seems to work :+1:

**Comment by ndom91** [comment]
You can also pin it to older commits via the Lazy option `commit`, as mentioned in that issue ^^

**Comment by artemlive** [comment]
> You can also pin it to older commits via the Lazy option `commit`, as mentioned in that issue ^^

It didn't work for me. 
I tried to pin [this](https://github.com/akinsho/bufferline.nvim/issues/895#issuecomment-2070772951). 

[Fork](https://github.com/Theyashsawarkar/bufferline.nvim) did the trick and it works.

**Comment by timofurrer** [comment]
Same issue here, but don't really want to run a fork 🤔

**Comment by GKSN** [comment]
Same problem here. Trying to solve, ...

**Comment by jiholland** [comment]
Had the same issue with neovim 0.10.0. Removing "version = *" (I use lazy and pretty much follow the installation example in README.md) and getting the latests updates (lazy sync) resolved this bug for me.

**Comment by cksidharthan** [comment]
> Had the same issue with neovim 0.10.0. Removing "version = *" (I use lazy and pretty much follow the installation example in README.md) and getting the latests updates (lazy sync) resolved this bug for me. Removing "version = *" made the plugin go from commit [f4b4b98](https://github.com/akinsho/bufferline.nvim/commit/f4b4b980ce88d546aeb69d11463187c8f716a1a1) to [73540cb](https://github.com/akinsho/bufferline.nvim/commit/73540cb95f8d95aa1af3ed57713c6720c78af915) which resolved the issue.

Yup this fixed it for me :)

**Comment by cxwx** [comment]
same issue

**Comment by timofurrer** [comment]
I've had `version = "v3.*"` in my config (Lazy) and removing it solved the issue for me.

**Comment by dssste** [comment]
```
use {
	'akinsho/bufferline.nvim',
	-- tag = "*",
	requires = 'nvim-tree/nvim-web-devicons'
}
```
It's `tag` for Packer in case some one is not aware.

**Comment by rshutt** [comment]
> > Had the same issue with neovim 0.10.0. Removing "version = *" (I use lazy and pretty much follow the installation example in README.md) and getting the latests updates (lazy sync) resolved this bug for me. Removing "version = *" made the plugin go from commit [f4b4b98](https://github.com/akinsho/bufferline.nvim/commit/f4b4b980ce88d546aeb69d11463187c8f716a1a1) to [73540cb](https://github.com/akinsho/bufferline.nvim/commit/73540cb95f8d95aa1af3ed57713c6720c78af915) which resolved the issue.
> 
> Yup this fixed it for me :)

This also fixed it for me!  It was weird.  I think I managed to upgrade nvim from brew while debugging because the previous version of nvim worked pinning `bufferline.nvim` to `v4.5.2` using `tag=` in the lazy config.  I then updated the brew and did a `brew upgrade`, and lo and behold, `bufferline.nvim` was broken again.  It was a head-scratcher for sure, but this all works now with 0.10.0 and the latest, non-version constrained `bufferline.nvim`

**Comment by DartMitai** [comment]
`E5108: Error executing lua ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/
ui.lua:119: Segments must be a list
stack traceback:
        [C]: in function 'assert'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:119: in func
tion 'get_component_size'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:480: in func
tion <...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:461>
        vim/shared.lua: in function 'get_components'
        ...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:56: in funct
ion <...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:54>`
version = '*'

**Comment by fabiodcorreia** [comment]
I have the same issue on mac with nvim 0.10

**Comment by rshutt** [comment]
> I have the same issue on mac with nvim 0.10

https://github.com/akinsho/bufferline.nvim/issues/903#issuecomment-2117066503

https://github.com/akinsho/bufferline.nvim/issues/903#issuecomment-2116061574

I didn't solve it, but I can confirm the fix works.

**Comment by icefed** [comment]
@akinsho can we make a new release?

**Comment by artemlive** [comment]
> Had the same issue with neovim 0.10.0. Removing "version = *" (I use lazy and pretty much follow the installation example in README.md) and getting the latests updates (lazy sync) resolved this bug for me. Removing "version = *" made the plugin go from commit [f4b4b98](https://github.com/akinsho/bufferline.nvim/commit/f4b4b980ce88d546aeb69d11463187c8f716a1a1) to [73540cb](https://github.com/akinsho/bufferline.nvim/commit/73540cb95f8d95aa1af3ed57713c6720c78af915) which resolved the issue.

This worked for me.

**Comment by kkoyung** [comment]
> Had the same issue with neovim 0.10.0. Removing "version = *" (I use lazy and pretty much follow the installation example in README.md) and getting the latests updates (lazy sync) resolved this bug for me. Removing "version = *" made the plugin go from commit [f4b4b98](https://github.com/akinsho/bufferline.nvim/commit/f4b4b980ce88d546aeb69d11463187c8f716a1a1) to [73540cb](https://github.com/akinsho/bufferline.nvim/commit/73540cb95f8d95aa1af3ed57713c6720c78af915) which resolved the issue.

This workaround works for me.
However, the background color of the bufferline turns into black. Any one have the same problem?

Before: (neovim-v0.9.5)
![screenshot_2024-05-20_12:19:28](https://github.com/akinsho/bufferline.nvim/assets/11942650/5112c64b-aec3-4e81-9fc9-5e50d38d25b0)

After: (neovim-v0.10.0)
![screenshot_2024-05-20_12:19:58](https://github.com/akinsho/bufferline.nvim/assets/11942650/87545e94-79c7-46db-8e71-f524bd1defdb)

Terminal emulator: foot (version: 1.17.2)
Colorscheme: [catppuccin/nvim](https://github.com/catppuccin/nvim)

**Comment by command-z-z** [comment]
> kkoyung

yes, it's same problem for me.

**Comment by akinsho** [comment]
Closing this and other v0.9.0 -> 0.10.0 migration issues as this has now been fixed on the latest release `4.6.0`

**Comment by kkoyung** [comment]
> > kkoyung
> 
> yes, it's same problem for me.

@command-z-z FYI, I just found that this issue has already filed at catppuccin/nvim repository. catppuccin/nvim#704

**Comment by pencilcheck** [comment]
FIxed by removing version as well

**Comment by iamshabell** [comment]
For people who will see it in the future, what worked for me is going to `butterfly.nvim' clone you have( in my mac it is `.local/`). You'll need to go these specific file: `/.local/share/nvim/lazy/bufferline.nvim/lua/bufferline/utils/init.lua`.

Then at the end of the file, there is this line:
```lua
M.is_list = vim.tbl_isarray or vim.tbl_islist
```

You'll need to replace to ( [73540cb](https://github.com/akinsho/bufferline.nvim/commit/73540cb95f8d95aa1af3ed57713c6720c78af915#diff-6151c9f0f245d99572a8f9ecb40c2145fda08f726b42e5c3f2b9fabe00e64b83R259) ):
```lua
if vim.fn.has("nvim-0.10") == 1 then
  M.is_list = vim.isarray or vim.islist
else
  M.is_list = vim.tbl_isarray or vim.tbl_islist
end

```

Then you'll have save it, and that worked for me :)

**Comment by phreakyphoenix** [comment]
perfect @iamshabell, tiny correction, add tilde at the start of the path
~/.local/share/nvim/lazy/bufferline.nvim/lua/bufferline/utils/init.lua

solved for me on mac m1 15.0.1
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1731601260
